api/src/main/scala/pl/touk/nussknacker/engine/api/MetaData.scala

```diff
@@ -46,16 +46,31 @@ sealed trait ScenarioSpecificData extends TypeSpecificData
 
 case class FragmentSpecificData(docsUrl: Option[String] = None) extends TypeSpecificData
 
+// TODO: rename to FlinkStreamMetaData
 case class StreamMetaData(parallelism: Option[Int] = None,
                           //we assume it's safer to spill state to disk and fix performance than to fix heap problems...
                           spillStateToDisk: Option[Boolean] = Some(true),
                           useAsyncInterpretation: Option[Boolean] = None,
-                          checkpointIntervalInSeconds: Option[Long] = None) extends ScenarioSpecificData {
+                          checkpointIntervalInSeconds: Option[Long] = None) extends BaseStreamMetaData[StreamMetaData] {
 
   def checkpointIntervalDuration  : Option[Duration]= checkpointIntervalInSeconds.map(Duration.apply(_, TimeUnit.SECONDS))
 
   def shouldUseAsyncInterpretation(implicit defaultValue: DefaultAsyncInterpretationValue) : Boolean = useAsyncInterpretation.getOrElse(defaultValue.value)
 
+  override def withParallelism(newParallelism: Option[Int]): StreamMetaData = copy(parallelism = newParallelism)
+
+}
+
+case class LiteStreamMetaData(parallelism: Option[Int] = None) extends BaseStreamMetaData[LiteStreamMetaData] {
+
+  override def withParallelism(newParallelism: Option[Int]): LiteStreamMetaData = copy(parallelism = newParallelism)
+
+}
+
+sealed trait BaseStreamMetaData[T <: BaseStreamMetaData[T]] extends ScenarioSpecificData {
```

Comment:
I wonder if we benefit from this trait (except for easier builder). Maybe for lite we won't have parallelism but sth else in the end?

Reply:
I don't see too many. One that I have in mind is that if we want to have similar expression of using streaming and streaming-lite, we should try to have consistent options for them. So if we will change the way how those number of workers is determined in streaming-lite, we should think about similar change in streaming. But we can do it also without type checking ... so I removed this parent class.